src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -603,6 +604,11 @@ private BoundExpression TransformCompoundAssignmentLHS(BoundExpression originalL
             return variableTemp;
         }
 
+        private static bool IsInvariantArray(TypeSymbol type)
+        {
+            return (type as ArrayTypeSymbol)?.ElementType?.IsSealed == true;
```

Comment:
What does this return for `T where T : struct`?

Reply:
We can treat `T where T : struct` arrays as nonvariant. 

I think `IsSealed` returns false for type parameters regardless of constraints, so it may need to be checked separately. Or not at all - it would be a rare and hardly testable scenario.

Reply:
Yes, `IsSealed` returns `false` for type parameters (and `IsMustInherit` returns `False` in VB), regardless of constraints. I've not been able to hit `IsInvariantArray` with an array of type parameters in C#. In VB I can hit `IsInvariantArray` with type parameters with `class` constraint so I've added such a test but I've left the implementations of `IsInvariantArray` as is, returning `false` for type parameters.